src/Language/DifferentialDatalog/NS.hs

```diff
@@ -147,6 +147,7 @@ ctxMVars d ctx =
          CtxRuleRAtom rl i        -> ([], map f2mf $ ruleRHSVars d rl i)
          CtxRuleRCond rl i        -> ([], map f2mf $ ruleRHSVars d rl i)
          CtxRuleRFlatMap rl i     -> ([], map f2mf $ ruleRHSVars d rl i)
+         CtxRuleRInspect rl i     -> ([], (map f2mf $ ruleRHSVars d rl i) ++ [("ddlog_weight", Just $ tBit 64), ("ddlog_timestamp", Just $ tBit 64)])
```

Comment:
The timestamp type is actually more complex.  For a non-recursive rule, the timestamp is indeed just u64. For a rule that is part of a recursive fragment, it is a pair of two timestamps: the global timestamp, which is the same as in the non-recursive case and the inner timestamp, which represents iteration of the fixed point computation.  There are two ways we can handle this. The RFC suggests:

```
typedef DDTimestamp = DDTSGlobal{epoch: u64}
                    | DDTSNested{epoch: u64, iteration: u64}
```

Now I am thinking that a better way would be to use different types depending on the context.  We can add a method to check if the rule is recursive and use the appropriate type based on that.